windows-02.md

```diff
@@ -67,18 +78,75 @@ Quota project "[RECENTLY_CREATED_PROJECT_ID]" was added to ADC which can be used
+Operation "operations/[VERY_LARGE_STRING]" finished successfully.
+%
+```
+
+Create the service account authorization keys.
+```bash
+% gcloud iam service-accounts keys create "%APPDATA%\gcloud\service_account_key.json" \
+    --iam-account=generic-service-account@[YOUR PROJECT ID].iam.gserviceaccount.com
+created key [VERY_LARGE_STRING] of type [json] as [$HOME/.config/gcloud/service_account_key.json] for [generic-service-account@[YOUR PROJECT ID].iam.gserviceaccount.com]
+%
+```
+
+Add the key to your `.bash_profile` through the command line.
+```bash
+% echo 'export GOOGLE_APPLICATION_CREDENTIALS="$APPDATA\gcloud\service_account_key.json"' >>~\.bash_profile
+%
+```
+
+Sync your terminal with the current state of your `.bash_profile`.
+```bash
+% source ~\.bash_profile
+%
+```
+
 We'll test this with the icj-project-template when the time comes. If you use OneDrive, you might have to use **Git Bash** for some steps instead of the terminal within VS Code.
 
-### Setting up the environment variable
-> There _may_ be some issues setting this up on Windows machines that use OneDrive. It has been successfully installed on a non-OneDrive setup using this method.
+### Setting up the environment variable for GitHub Codespaces
+This environment variable will be used when you are accessing this project through [GitHub Codespaces](https://github.com/features/codespaces).
+We will be following the process shown [here](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-encrypted-secrets-for-your-codespaces#adding-a-secret).
+
+**NOTE:** It is absolutely imperative that you DO NOT commit the contents of `service_account_key.json` to your repository at all. If someone else were able to see the contents of that file, they could execute any action that service account has in its abilities.
+Since `service_account_key.json` is in the `.gitignore` file, you should not be able to check it in, but it is important to remember that for the sake of transparency.
 
-We are setting this environment variable to authenticate ourselves to Google using the information in the json file you just created.
+Step-specific information:
+4. The "Name" of the secret will be `GOOGLE_CREDENTIALS`.
+5. The "Value" of the secret will be the contents of the `$APPDATA\gcloud\service_account_key.json` file.
+6. The repository that you will give access to it will be the name of the repository you are using in the Codespace, presumably `icj-project-rig`.
 
-1. Open a new Git Bash prompt.
-2. Enter `code .bash_profile` to open your `.bash_profile` file in VS Code. You should see some stuff there already from other configurations. (Holler if you don't as that means you are likely in the wrong file.)
+When you open your Codespace, you will run the following command, and you should be able to get to work quickly:
 ```bash
-# Google Auth
-set GOOGLE_APPLICATION_CREDENTIALS="$APPDATA\gcloud\application_default_credentials.json"
+$ npm run codespace-google-auth
```

Comment:
I saved the previous bash command which could be a bit cumbersome, to a simple npm command: 
```bash
@michplunkett ➜ /workspaces/icj-project-rig (main) $ npm run codespace-google-auth

> icj-project-template@1.1.0 codespace-google-auth
> printenv GOOGLE_CREDENTIALS > service_account_key.json && export GOOGLE_APPLICATION_CREDENTIALS=service_account_key.json

@michplunkett ➜ /workspaces/icj-project-rig (main) $ gulp fetch
`import sass from 'sass'` is deprecated.
Please use `import * as sass from 'sass'` instead.
[03:49:48] Using gulpfile /workspaces/icj-project-rig/gulpfile.js
[03:49:48] Starting 'fetch'...
[03:49:48] Finished 'fetch' after 4.08 ms
Downloaded `library` (1RgMhjtkXlbbf9uzSzy_xPRKwxcVZIZqVytgM_JoU4E4)
Downloaded `bookstores` (1gDwO-32cgpBDn_0niV0iu6TqQTaRDr4nmSqnT53magY)
@michplunkett ➜ /workspaces/icj-project-rig (main) $ gulp format
`import sass from 'sass'` is deprecated.
Please use `import * as sass from 'sass'` instead.
[03:49:55] Using gulpfile /workspaces/icj-project-rig/gulpfile.js
[03:49:55] Starting 'format'...
[03:49:55] Finished 'format' after 396 ms
@michplunkett ➜ /workspaces/icj-project-rig (main) $